File: java/0001.0_Two_Sum.java
/*
执行用时：1 ms, 在所有 Java 提交中击败了99.32% 的用户
内存消耗：41.2 MB, 在所有 Java 提交中击败了76.82% 的用户
通过测试用例：57 / 57
*/

class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] indices = new int[2];
        Map<Integer, Integer> m = new HashMap<Integer, Integer>();
        for (int i = 0; i < nums.length; i++) {
            if (m.containsKey(target - nums[i])) {
                indices[0] = m.get(target - nums[i]);
                indices[1] = i;
                return indices;
            }
            m.put(nums[i], i);
        }
        return indices;
    }
}


/*
执行用时：2 ms, 在所有 Java 提交中击败了71.80% 的用户
内存消耗：41.6 MB, 在所有 Java 提交中击败了28.91% 的用户
通过测试用例：57 / 57
*/
import java.util.HashMap;
import java.util.Map;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] indices = new int[2];
        Map<Integer, Integer> m = new HashMap<Integer, Integer>();
        for (int i = 0; i < nums.length; i++) {
            if (m.containsKey(nums[i])) {
                indices[0] = m.get(nums[i]);
                indices[1] = i;
                return indices;
            }
            m.put(target - nums[i], i);
        }
        return indices;
    }
}

File: csharp/0001.0_Two_Sum.cs
/*
执行用时：128 ms, 在所有 C# 提交中击败了96.54% 的用户
内存消耗：43.7 MB, 在所有 C# 提交中击败了24.49% 的用户
通过测试用例：57 / 57
*/

public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        Dictionary<int, int> d = new Dictionary<int, int>();
        for (int i = 0; i < nums.Length; ++i)
        {
            if (d.ContainsKey(nums[i])) return new int[]{d[nums[i]], i};
            if (!d.ContainsKey(target - nums[i])) d.Add(target - nums[i], i);
        }
        return null;
    }
}
